File: push/src/main/java/com/xiaomi/push/service/JobSchedulerTaskWrapper.java
package com.xiaomi.push.service;

final class JobSchedulerTaskWrapper {
    boolean cancelled;
    JobScheduler.Job job;
    final Object lock = new Object();
    private long scheduledTime;
    int type;
    long when;

    boolean cancel() {
        boolean z;
        synchronized (this.lock) {
            z = !this.cancelled && this.when > 0;
            this.cancelled = true;
        }
        return z;
    }

    long getWhen() {
        long j;
        synchronized (this.lock) {
            j = this.when;
        }
        return j;
    }

    long scheduledExecutionTime() {
        long j;
        synchronized (this.lock) {
            j = this.scheduledTime;
        }
        return j;
    }

    void setScheduledTime(long j) {
        synchronized (this.lock) {
            this.scheduledTime = j;
        }
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/push/service/JobSchedulerTaskWrapper.kt
package com.xiaomi.push.service

class JobSchedulerTaskWrapper {
    @JvmField
    var cancelled: Boolean = false

    @JvmField
    var job: JobScheduler.Job? = null

    @JvmField
    val lock: Any = Any()

    private var scheduledTime: Long = 0L

    @JvmField
    var type: Int = 0

    @JvmField
    var `when`: Long = 0L

    fun cancel(): Boolean {
        synchronized(lock) {
            val shouldCancel = !cancelled && `when` > 0
            cancelled = true
            return shouldCancel
        }
    }

    fun getWhen(): Long {
        synchronized(lock) {
            return `when`
        }
    }

    fun scheduledExecutionTime(): Long {
        synchronized(lock) {
            return scheduledTime
        }
    }

    fun setScheduledTime(value: Long) {
        synchronized(lock) {
            scheduledTime = value
        }
    }
}
